doc/releases/changelog-dev.md

```diff
@@ -9,6 +9,7 @@
 * PennyLane is now compatible with NumPy 2.0.
   [(#6061)](https://github.com/PennyLaneAI/pennylane/pull/6061)
   [(#6258)](https://github.com/PennyLaneAI/pennylane/pull/6258)
+  [(#6255)](https://github.com/PennyLaneAI/pennylane/pull/6255)
```

Comment:
Do we want to mention the compatibility with JAX 0.4.28 somewhere? :)

Reply:
> Also, do we want to change requirements files?

It seems to me that all external interfaces such as `tensorflow`, `jax`, `torch`, etc. are not mentioned in the requirements files.

> Do we want to mention the compatibility with JAX 0.4.28 somewhere?

I agree. Since we changed the source code (and marked with `xfail` some tests), it's probably best to create a separate changelog entry for this PR (although the context is upgrading PL to work with NumPy 2). I'll push a commit later 👍